# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1375880935 14400
#      Wed Aug 07 09:08:55 2013 -0400
# Node ID 755ec7bce733e094a325fa4d8eac301d4bb348da
# Parent  85e55da6140902c85e3b04ebb2725082524a9aad
corefcn/graphics.cc: Fix memory leak in uicontrol::properties::update_text_extent.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -7522,16 +7522,17 @@ uicontrol::properties::update_text_exten
   elt = text_parser_none ().parse (get_string_string ());
 #ifdef HAVE_FONTCONFIG
   text_renderer.set_font (get_fontname (),
                           get_fontweight (),
                           get_fontangle (),
                           get_fontsize ());
 #endif
   box = text_renderer.get_extent (elt, 0);
+  delete elt;
 
   Matrix ext (1, 4, 0.0);
 
   // FIXME: also handle left and bottom components
 
   ext(0) = ext(1) = 1;
   ext(2) = box(0);
   ext(3) = box(1);
